feat(effects): resubscribe to effects on error (#1881)
ngrx/platform: projects/ngrx.io/content/guide/effects/lifecycle.md

}
</code-example>

### Resubscribe on Error

> Starting from NgRx V8 when error happens in the effect's main stream it is

Reviewer: ```suggestion
Starting with version 8, when an error happens in the effect's main stream it is
```

> reported using Angular's `ErrorHandler` and the source effect is 

Reviewer: ```suggestion
reported using Angular's `ErrorHandler`, and the source effect is 
```

> **automatically** resubscribe (instead of completeling), so it continues to 

Reviewer: ```suggestion
**automatically** resubscribed to (instead of completing), so it continues to 
```

Reviewer 2: @alex-okrushko You missed this one 😉

> listen to any new dispatched Actions.

Reviewer: ```suggestion
listen to all dispatched Actions.
```

> Generally, errors should be handled by users and operators such as `mapToAction`

Reviewer: ```suggestion
Generally, errors should be handled by users, and operators such as `mapToAction`
```

Reviewer 2: @alex-okrushko You missed this one 😉

should make it easier to do. However, for the cases where errors were missed, 

> this new behavior adds additional safety net.

Reviewer: ```suggestion
this new behavior adds an additional safety net.
```

> In some cases where particular RxJS operators are used the new behavior might

Reviewer: ```suggestion
In some cases where particular RxJS operators are used, the new behavior might
```

> produce unexpected results. For example, if the `startWith` operator is within 

Reviewer: ```suggestion
produce unexpected results. For example, if the `startWith` operator is within the
```

effect's pipe then it will be triggered again.

> To remove resubscriptions add `{resubscribeOnError: false}` to `createEffect` 

Reviewer: ```suggestion
To disable resubscriptions add `{resubscribeOnError: false}` to the `createEffect` 
```

metadata (second argument).

<code-example header="disable-resubscribe.effects.ts">
import { Injectable } from '@angular/core';
import { Actions, Effect, ofType } from '@ngrx/effects';
import { tap } from 'rxjs/operators';

@Injectable()

> export class ErrorFreeEffects {

Reviewer: I think we need a different example here. This infinite loop has bitten us enough times.

Author: Provided what I hope is a better example

}
</code-example>

### Resubscribe on Error

> Starting with version 8, when an error happens in the effect's main stream it is

Reviewer 2: I think we can leave out `Starting with version 8` as these are the docs for v8?

Author: Yes, but I'd rather highlight that it's a change from previous behavior. I find that beneficial. We can remove it by v9 :)
